### DISCUSSION TITLE
Filtering with `height` --flag at LCD queries

### DISCUSSION THREAD

**Discussion by RaulBernal** [cosmos/community#9] [Chain Builders (Cosmos SDK)] [answered]
Anyone could realize why when I ask for list all the`bitcannaid` registries by LCD it doesn't affect the `?height=xxxxx` filter?

I solved it for the CLI (https://github.com/BitCannaGlobal/bcna/commit/2e90dc3541eb0f0ebedaa5154e87d8faf3a6c771) but can't realize where to fix for LCD.
This doesn't works: https://lcd-devnet-6.bitcanna.io/BitCannaGlobal/bcna/bcna/bitcannaid?height=1  (it shouldn't show registries)

I guess something is missing with the context here, in example:
https://github.com/BitCannaGlobal/bcna/blob/main/x/bcna/keeper/bitcannaid.go  
or https://github.com/BitCannaGlobal/bcna/blob/main/x/bcna/keeper/query_bitcannaid.go 🤷‍♂️ 

BTW I saw a lot of custom modules in the ecosystem for different chains lacking this issue.
In example, the liquidity module for `gaiad` also do:

`gaiad q liquidity pools --height 1` produces the same output than `gaiad q liquidity pools --height 10638712` than `gaiad q liquidity pools`

**The same counts for the LCD**. I know how to fix the CLI here https://github.com/BitCannaGlobal/bcna/blob/4d50e3da1f53363f0791cab885dd64892b70129d/x/bcna/client/cli/query_bitcannaid.go#L19
but not the LCD reply.

Issue [created](https://github.com/Gravity-Devs/liquidity/issues/28)

**Comment by RaulBernal** [comment]
any hint @julienrbrt?

**Reply by julienrbrt** [reply]
I have not looked at the code yet, but just reading the post, I wouldn't know why.
Let me ask someone. I can dive into it a bit later otherwise.

**Comment by facundomedica** [comment, marked as answer]
The height must be passed in as an HTTP header: `x-cosmos-block-height`.

`curl 'https://lcd-devnet-6.bitcanna.io/BitCannaGlobal/bcna/bcna/bitcannaid' -H 'x-cosmos-block-height: 123456'`
